### PR TITLE
Added Apprentice Riding (33388) spell to Death Knight playercreateinf…

### PR DESCRIPTION
**Pull request by zeb139**
I couldn't find any documentation of Death Knights already having Expert Riding without buying the training in retail, so I only fixed the issue with Apprentice Riding.
^ Please correct me if I'm mistaken.

**Changes proposed:**
-  Added Apprentice Riding (33388) to 'playercreateinfo_spells' table in 'world' and 'characters' databases for Death Knights of all Races.

**Target branch(es):**  master

**Issues addressed:** Updates #639 

**Tests performed:** Tested in-game and working.

### CODE REVIEW

**Comment by Nefertumm** [comment]
Please read:
[https://github.com/azerothcore/azerothcore-wotlk/wiki/Dealing-with-SQL-files](https://github.com/azerothcore/azerothcore-wotlk/wiki/Dealing-with-SQL-files)

plus, you have syntax errors. I will review your pr in order you can fix them.

File: data/sql/updates/pending_db_world/rev_1520902066724046200.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM `world`.`playercreateinfo_spell` where `Note`='Apprentice Riding';
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1520902066724046200.sql]:
```sql
DELETE FROM `playercreateinfo_spell` WHERE `Note` = 'Apprentice Riding';
```

Not sure if this is the right condition to do this delete.

File: data/sql/updates/pending_db_world/rev_1520902066724046200.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+INSERT INTO `world`.`playercreateinfo_spell` (`racemask`, `classmask`, `Spell`, `Note`) VALUES (0, 32, 33388, 'Apprentice Riding');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1520902066724046200.sql]:
```sql
INSERT INTO `playercreateinfo_spell` (`racemask`, `classmask`, `Spell`, `Note`) VALUES (0, 32, 33388, 'Apprentice Riding');
```

**Comment by zeb139** [comment]
Thanks for the feedback!
> Not sure if this is the right condition to do this delete.  

Oh yeah! "What if the language changes?" I'll check the Spell id instead.

I can't seem to figure out the SQL syntax issue. Using an online validator, both of these statements are fine by themselves, but together they create an error. Do you know why this is?
```sql
DELETE FROM 
    `playercreateinfo_spell` 
WHERE 
    `Spell`=33388;
INSERT INTO 
    `playercreateinfo_spell` (`racemask`, `classmask`, `Spell`, `Note`) 
VALUES 
    (0, 32, 33388, 'Apprentice Riding');
```

**Comment by talamortis** [comment]
Travis should work correctly now, reason why it failed as you was putting the sql in the character database where it should be in the world database only.

**Comment by BarbzYHOOL** [comment]
Someone confirms this is blizzlike?

**Comment by Knindzagxg** [comment]
Yeah, this is Blizzlike.
Death Knights are supposed to have it, due mount quest.

**Comment by BarbzYHOOL** [comment]
Ok, but now it's missing that kind of line

```
 INSERT INTO version_db_world (`sql_rev`) VALUES ('1528460337456239208'); 
```

**Comment by BarbzYHOOL** [comment]
Ok I added it, not sure if it is right though.

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL @Nefertumm 

Is this ready to be tested and merged?

**Comment by BarbzYHOOL** [comment]
Seems good but wait for someone else's opinion :P

**Comment by wetbrownsauce** [comment]
Tested, works fine, however the PR still merges the old sql using the sketchy
```
DELETE FROM `playercreateinfo_spell` WHERE `Note` = 'Apprentice Riding';
```
instead of the updated and safer:
```
DELETE FROM 
    `playercreateinfo_spell` 
WHERE 
    `Spell`=33388 AND `racemask`=0 AND `classmask`=32;
```
I would recommend swapping those statements before merging.  


That aside, results of test:
- Death Knights (and only death knights) now start with both apprentice and journeyman riding so now neither are available to re-learn at a riding trainer.  

Looks good!  



If anyone is questioning whether or not this is blizzlike, I dug up [this forum thread](https://us.battle.net/forums/en/wow/topic/2140190720) from 2011. 

Quote from a forum MVP:
>You'll get your epic land mount from a quest **as well as the training.** Any riding tiers beyond that/mounts you'll have to pay for.

**Comment by BarbzYHOOL** [comment]
thank you @wetbrownsauce !!

Though I don't get why there is a "delete from" query??

Is it because of this in the wiki "use INSERT in order to insert new rows only, but be sure to avoid import errors using DELETE before INSERT" ?? which import error ? there is no existing row prior to the pr

**Comment by BarbzYHOOL** [comment]
@zeb139 thank you!

**Pull request merged**